build_cell_ontology_dict.R

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env Rscript 
+
+# This script accepts a collection of (condensed) SDRF files as input and returns a 
+# dictionary containing cell labels as keys and ontology terms as values 
+# for all cells in corresponding SDRF files. In case a cell does not have 
+# ontology mapping, label will map to NA value. 
+
+suppressPackageStartupMessages(require(optparse))
+suppressPackageStartupMessages(require(workflowscriptscommon))
+
+option_list = list(
+    make_option(
+        c("-i", "--input-dir"),
+        action = "store",
+        default = NA,
+        type = 'character',
+        help = 'Path to the directory with condensed SDRF files'
+    ),
+    make_option(
+        c("-k", "--condensed-sdrf"),
+        action = "store_true",
+        default = FALSE,
+        type = 'logical',
+        help = 'Boolean: is the provided SDRF file in a condensed form? Default: TRUE'
+    ),
+    make_option(
+        c("-b", "--barcode-col-name"),
+        action = "store",
+        default = "id",
+        type = 'character',
+        help = 'Name of the barcode column in SDRF files (must be identical across all files)'
+    ),
+    make_option(
+        c("-l", "--cell-label-col-name"),
+        action = "store",
+        default = "cell.type",
+        type = 'character',
+        help = 'Name of the cell label column in SDRF files (must be identical across all files)'
+    ),
+    make_option(
+        c("-c", "--cell-ontology-col-name"),
+        action = "store",
+        default = "cell.type.ontology",
+        type = 'character',
+        help = 'Name of the cell ontology terms column in SDRF files (must be identical across all files)'
+    ),
+    make_option(
+        c("-o", "--output-dict-path"),
+        action = "store",
+        default = NA,
+        type = 'character',
+        help = 'Output path for serialised object containing the dictionary'
+    ) 
+)
+
+# parse arguments 
+opt = wsc_parse_args(option_list, mandatory = c("input_dir", "output_dict_path"))
+# source function definitions 
+p = system("which cell_types_utils.R", intern = TRUE)
```

Comment:
I can see why you're doing this, but I'm not keen- the PATH could be changed and would break this. Could you rather source the utilities relative to this script? That's messy in R too, but possible. The best I came up with is:

```
dirname(strsplit(commandArgs()[grep('--file=', commandArgs())], '=')[[1]][2])
```
Feel free to use a neater solution to do the equivalent if you can find one.

Ultimately we could make the utilities an R package as we did for r-workflowscriptscommon. Or we could make this scripts package an R package itself as Ni did for monocle-cli- but that's a job for later.